infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -111,23 +117,24 @@ public void handle(@Nonnull CorfuPayloadMsg<OrchestratorMsg> msg,
                 query(msg, ctx, r);
                 break;
             case ADD_NODE:
-                workflow = new AddNodeWorkflow((AddNodeRequest) orchReq.getRequest());
+                workflow = new AddNodeWorkflow((AddNodeRequest) orchReq.getRequest(), streamLog);
                 dispatch(workflow, msg, ctx, r);
                 break;
             case REMOVE_NODE:
                 workflow = new RemoveNodeWorkflow((RemoveNodeRequest) orchReq.getRequest());
                 dispatch(workflow, msg, ctx, r);
                 break;
             case HEAL_NODE:
-                workflow = new HealNodeWorkflow((HealNodeRequest) orchReq.getRequest());
+                workflow = new HealNodeWorkflow((HealNodeRequest) orchReq.getRequest(), streamLog);
                 dispatch(workflow, msg, ctx, r);
                 break;
             case FORCE_REMOVE_NODE:
                 workflow = new ForceRemoveWorkflow((ForceRemoveNodeRequest) orchReq.getRequest());
                 dispatch(workflow, msg, ctx, r);
                 break;
             case RESTORE_REDUNDANCY_MERGE_SEGMENTS:
-                workflow = new RestoreRedundancyMergeSegmentsWorkflow((RestoreRedundancyMergeSegmentsRequest) orchReq.getRequest());
+                workflow = new RestoreRedundancyMergeSegmentsWorkflow(
```

Comment:
I don't see this as necessary. Why not conflate it with `HEAL_NODE,` they only differ in the layout changes, which are small and not worth have a different code path. I think its cleaner to make `HealNodeToLayout` idempotent.

Reply:
As per conversation with Zeeshan, this workflow is necessary. The heal can fail even after the retries. We can have the situation, where the layout does not contain the node in the unresponsive list but the redundancy is still not fully restored. This workflow makes sure that the node's redundancy is restored and the segments are merged if possible.